src/discord/commands/create_event.rs

```diff
@@ -0,0 +1,92 @@
+use std::str::FromStr;
+
+use crate::{calendar::Client as CalendarClient, Error};
+use chrono::{Datelike, Days, NaiveDate, Utc};
+use google_calendar3::api::{Event, EventDateTime};
+use serenity::all::{
+    CommandOptionType, Context, CreateCommand, CreateCommandOption, GuildId, Permissions,
+    ResolvedOption, ResolvedValue,
+};
+use tracing::{instrument, warn};
+
+use super::MessageResult;
+
+#[instrument]
+pub async fn run(
+    ctx: &Context,
+    guild_id: &GuildId,
+    options: &[ResolvedOption<'_>],
+) -> MessageResult {
+    let Some(ResolvedOption {
+        value: ResolvedValue::String(label),
+        ..
+    }) = options.first()
+    else {
+        return Err(Error::MissingParameter("label".into()));
+    };
+
+    let date = match options.get(1) {
+        Some(ResolvedOption {
+            value: ResolvedValue::String(date),
+            ..
+        }) => {
+            let year = Utc::now().year();
+            let date = format!("{year}-{date}");
+            NaiveDate::from_str(&date)?
+        }
+        _ => Utc::now().date_naive(),
+    };
+
+    let event = Event {
+        summary: Some(label.to_string()),
+        start: Some(EventDateTime {
+            date: Some(date),
+            ..Default::default()
+        }),
+        end: Some(EventDateTime {
+            date: Some(
+                date.checked_add_days(Days::new(1))
+                    .expect("Out of range days"),
+            ),
+            ..Default::default()
+        }),
+        recurrence: Some(vec!["RRULE:FREQ=YEARLY".into()]),
+        ..Default::default()
+    };
```

Comment:
Oh, I see that there's a yearly recurrence by default, so most likely it's meant for something like birthdays or memorable dates. :)

Wanted to suggest adding a time label too, but for this use case it's not needed.